### ISSUE TITLE
[Token] Spacing (semantic) modifications after Figma updates (4th of October)

### ISSUE THREAD

**Issue by julien-deramond** [Orange-OpenSource/ouds-ios#156]
## Description

Some modifications have been done on Figma's side. Here's a diff showing what's been modified the 4th of October by the designers.

```diff
diff --git a/build/ios/semantic/SpaceSemanticTokens.swift b/build/ios/semantic/SpaceSemanticTokens.swift
index 5dad41c..df084f7 100644
--- a/build/ios/semantic/SpaceSemanticTokens.swift
+++ b/build/ios/semantic/SpaceSemanticTokens.swift
@@ -12,36 +12,36 @@
 // 
 
 public enum SpaceSemanticTokens {
-    public static let spaceScaledTabletTallest = dimension600
-    public static let spaceScaledTabletTaller = dimension500
-    public static let spaceScaledTabletTall = dimension400
-    public static let spaceScaledTabletSpacious = dimension700
-    public static let spaceScaledTabletSmash = dimension50
-    public static let spaceScaledTabletShortest = dimension100
-    public static let spaceScaledTabletShorter = dimension150
-    public static let spaceScaledTabletShort = dimension200
-    public static let spaceScaledTabletNone = dimension0
-    public static let spaceScaledTabletMedium = dimension300
-    public static let spaceScaledMobileTallest = dimension500
-    public static let spaceScaledMobileTaller = dimension400
-    public static let spaceScaledMobileTall = dimension300
-    public static let spaceScaledMobileSpacious = dimension600
-    public static let spaceScaledMobileSmash = dimension25
-    public static let spaceScaledMobileShortest = dimension50
-    public static let spaceScaledMobileShorter = dimension100
-    public static let spaceScaledMobileShort = dimension150
-    public static let spaceScaledMobileNone = dimension0
-    public static let spaceScaledMobileMedium = dimension200
-    public static let spaceScaledDesktopTallest = dimension700
-    public static let spaceScaledDesktopTaller = dimension600
-    public static let spaceScaledDesktopTall = dimension500
-    public static let spaceScaledDesktopSpacious = dimension800
-    public static let spaceScaledDesktopSmash = dimension50
-    public static let spaceScaledDesktopShortest = dimension100
-    public static let spaceScaledDesktopShorter = dimension200
-    public static let spaceScaledDesktopShort = dimension300
-    public static let spaceScaledDesktopNone = dimension0
-    public static let spaceScaledDesktopMedium = dimension400
+    public static let spaceScaledTallestTablet = dimension600
+    public static let spaceScaledTallestMobile = dimension500
+    public static let spaceScaledTallestDesktop = dimension700
+    public static let spaceScaledTallerTablet = dimension500
+    public static let spaceScaledTallerMobile = dimension400
+    public static let spaceScaledTallerDesktop = dimension600
+    public static let spaceScaledTallTablet = dimension400
+    public static let spaceScaledTallMobile = dimension300
+    public static let spaceScaledTallDesktop = dimension500
+    public static let spaceScaledSpaciousTablet = dimension700
+    public static let spaceScaledSpaciousMobile = dimension600
+    public static let spaceScaledSpaciousDesktop = dimension800
+    public static let spaceScaledSmashTablet = dimension50
+    public static let spaceScaledSmashMobile = dimension25
+    public static let spaceScaledSmashDesktop = dimension50
+    public static let spaceScaledShortestTablet = dimension100
+    public static let spaceScaledShortestMobile = dimension50
+    public static let spaceScaledShortestDesktop = dimension100
+    public static let spaceScaledShorterTablet = dimension150
+    public static let spaceScaledShorterMobile = dimension100
+    public static let spaceScaledShorterDesktop = dimension200
+    public static let spaceScaledShortTablet = dimension200
+    public static let spaceScaledShortMobile = dimension150
+    public static let spaceScaledShortDesktop = dimension300
+    public static let spaceScaledNoneTablet = dimension0
+    public static let spaceScaledNoneMobile = dimension0
+    public static let spaceScaledNoneDesktop = dimension0
+    public static let spaceScaledMediumTablet = dimension300
+    public static let spaceScaledMediumMobile = dimension200
+    public static let spaceScaledMediumDesktop = dimension400
     public static let spaceFixedTallest = dimension500
     public static let spaceFixedTallest = dimension500
     public static let spaceFixedTaller = dimension400
     public static let spaceFixedTall = dimension300
@@ -66,22 +66,24 @@ public enum SpaceSemanticTokens {
     public static let spaceRowGapShort = dimension9xs
     public static let spaceRowGapNone = dimensionNone
     public static let spaceRowGapMedium = dimension8xs
-    public static let spacePaddingInlineWithIconTallest = dimension3xs
-    public static let spacePaddingInlineWithIconTaller = dimension5xs
-    public static let spacePaddingInlineWithIconTall = dimension7xs
-    public static let spacePaddingInlineWithIconShorter = dimension10xs
-    public static let spacePaddingInlineWithIconShort = dimension9xs
+    public static let spacePaddingInlineWithIconTallest = dimension2xs
+    public static let spacePaddingInlineWithIconTaller = dimension3xs
+    public static let spacePaddingInlineWithIconTall = dimension5xs
+    public static let spacePaddingInlineWithIconShortest = dimension10xs
+    public static let spacePaddingInlineWithIconShorter = dimension9xs
+    public static let spacePaddingInlineWithIconShort = dimension8xs
     public static let spacePaddingInlineWithIconNone = dimensionNone
-    public static let spacePaddingInlineWithIconMedium = dimension8xs
+    public static let spacePaddingInlineWithIconMedium = dimension7xs
     public static let spacePaddingInlineWithArrowTallest = dimension3xs
-    public static let spacePaddingInlineWithArrowTaller = dimension5xs
-    public static let spacePaddingInlineWithArrowTall = dimension7xs
-    public static let spacePaddingInlineWithArrowShorter = dimension10xs
-    public static let spacePaddingInlineWithArrowShort = dimension9xs
+    public static let spacePaddingInlineWithArrowTaller = dimension4xs
+    public static let spacePaddingInlineWithArrowTall = dimension6xs
+    public static let spacePaddingInlineWithArrowShortest = dimension10xs
+    public static let spacePaddingInlineWithArrowShorter = dimension9xs
+    public static let spacePaddingInlineWithArrowShort = dimension8xs
     public static let spacePaddingInlineWithArrowNone = dimensionNone
-    public static let spacePaddingInlineWithArrowMedium = dimension8xs
-    public static let spacePaddingInlineTallest = dimensionMd
-    public static let spacePaddingInlineTaller = dimensionXs
+    public static let spacePaddingInlineWithArrowMedium = dimension7xs
+    public static let spacePaddingInlineTallest = dimensionXs
+    public static let spacePaddingInlineTaller = dimension2xs
     public static let spacePaddingInlineTall = dimension3xs
     public static let spacePaddingInlineShorter = dimension9xs
     public static let spacePaddingInlineShort = dimension7xs
```

Sorry, the parser is not yet able to handle transitivity (that's why there are values such as "dimension7xs", etc.). Here's the mapping table to help you understand the modifications based on what you have in the library:

```swift
public static let dimensionXs = dimension400
public static let dimensionXl = dimension600
public static let dimensionSm = dimension450
public static let dimensionNone = dimension0
public static let dimensionMd = dimension500
public static let dimensionLg = dimension550
public static let dimension9xs = dimension50
public static let dimension9xl = dimension1400
public static let dimension8xs = dimension75
public static let dimension8xl = dimension1200
public static let dimension7xs = dimension100
public static let dimension7xl = dimension1000
public static let dimension6xs = dimension150
public static let dimension6xl = dimension900
public static let dimension5xs = dimension200
public static let dimension5xl = dimension800
public static let dimension4xs = dimension250
public static let dimension4xl = dimension750
public static let dimension3xs = dimension300
public static let dimension3xl = dimension700
public static let dimension2xs = dimension350
public static let dimension2xl = dimension650
public static let dimension11xl = dimension1800
public static let dimension10xs = dimension25
public static let dimension10xl = dimension1600
```

**Comment by pylapp** [comment]
@julien-deramond FYI here is the output of the update Swift files :)

The tokens car refer to other dimension semantic token, no need of transitivity

```swift
    @objc open var spacePaddingInlineWithIconNone: SpacingPaddingInlineSemanticToken { dimensionNone }
    @objc open var spacePaddingInlineWithIconShortest: SpacingPaddingInlineSemanticToken { dimension10XS }
    @objc open var spacePaddingInlineWithIconShorter: SpacingPaddingInlineSemanticToken { dimension9XS }
    @objc open var spacePaddingInlineWithIconShort: SpacingPaddingInlineSemanticToken { dimension8XS }
    @objc open var spacePaddingInlineWithIconMedium: SpacingPaddingInlineSemanticToken { dimension7XS }
    @objc open var spacePaddingInlineWithIconTall: SpacingPaddingInlineSemanticToken { dimension5XS }
    @objc open var spacePaddingInlineWithIconTaller: SpacingPaddingInlineSemanticToken { dimension3XS }
    @objc open var spacePaddingInlineWithIconTallest: SpacingPaddingInlineSemanticToken { dimension2XS }

    @objc open var spacePaddingInlineWithArrowNone: SpacingPaddingInlineSemanticToken { dimensionNone }
    @objc open var spacePaddingInlineWithArrowShortest: SpacingPaddingInlineSemanticToken { dimension10XS }
    @objc open var spacePaddingInlineWithArrowShorter: SpacingPaddingInlineSemanticToken { dimension9XS }
    @objc open var spacePaddingInlineWithArrowShort: SpacingPaddingInlineSemanticToken { dimension8XS }
    @objc open var spacePaddingInlineWithArrowMedium: SpacingPaddingInlineSemanticToken { dimension7XS }
    @objc open var spacePaddingInlineWithArrowTall: SpacingPaddingInlineSemanticToken { dimension6XS }
    @objc open var spacePaddingInlineWithArrowTaller: SpacingPaddingInlineSemanticToken { dimension4XS }
    @objc open var spacePaddingInlineWithArrowTallest: SpacingPaddingInlineSemanticToken { dimension3XS }
```

To reduce the number of variables, we gather both "mobile" (i.e. compact) and "tablet" (i.e. variants regular)

```swift
    @objc open var spaceScaledNone: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension0, regular: DimensionRawTokens.dimension0) }
    @objc open var spaceScaledSmash: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension25, regular: DimensionRawTokens.dimension50) }
    @objc open var spaceScaledShortest: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension50, regular: DimensionRawTokens.dimension100) }
    @objc open var spaceScaledShorter: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension100, regular: DimensionRawTokens.dimension150) }
    @objc open var spaceScaledShort: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension150, regular: DimensionRawTokens.dimension200) }
    @objc open var spaceScaledMedium: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension200, regular: DimensionRawTokens.dimension300) }
    @objc open var spaceScaledTall: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension300, regular: DimensionRawTokens.dimension400) }
    @objc open var spaceScaledTaller: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension400, regular: DimensionRawTokens.dimension500) }
    @objc open var spaceScaledTallest: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension500, regular: DimensionRawTokens.dimension600) }
    @objc open var spaceScaledSpacious: MultipleSpacingTokens { MultipleSpacingTokens(compact: DimensionRawTokens.dimension600, regular: DimensionRawTokens.dimension700) }
```

**Comment by pylapp** [comment]
ℹ️ Released with version [v0.4.0](https://github.com/Orange-OpenSource/ouds-ios/releases/tag/0.4.0) 🚀
